### PR TITLE
Feat/string mocker improvement

### PR DESCRIPTION
**Pull request by sixaphone**

### CODE REVIEW

File: src/samplers/string.js [hunk 1]
```diff
@@ -60,6 +62,22 @@ function uriSample() {
   return 'http://example.com';
 }
 
+function byteSample() {
+  return Buffer.from('Hello Mock').toString('base64');
```
Comment (derevnjuk) [inline src/samplers/string.js]:
```suggestion
  return Buffer.from(faker.lorem.word()).toString('base64');
```

File: src/samplers/string.js [hunk 2]
```diff
@@ -60,6 +62,22 @@ function uriSample() {
   return 'http://example.com';
```
Comment (derevnjuk) [inline src/samplers/string.js]:
```suggestion
   return faker.internet.url();
```

Comment (derevnjuk) [reply src/samplers/string.js]: Please, use random data

**Review by derevnjuk — CHANGES_REQUESTED** [review]

**Comment by derevnjuk** [comment]
@sixaphone Maybe we try to add support `not`. What do you think?

**Comment by sixaphone** [comment]
Sure i think i can do that no prob

File: src/samplers/string.js [hunk 3]
```diff
@@ -95,11 +105,7 @@ const stringFormats = {
 };
 
 export function sampleString(schema) {
-  if (schema.pattern !== undefined) {
-    return stringFormats['pattern'](schema.pattern);
-  } else {
-    let format = schema.format || 'default';
+    let format = schema.pattern !== undefined ? 'pattern' : schema.format || 'default';
```
Comment (derevnjuk) [inline src/samplers/string.js]: What  happens, if `pattern` will be `null`?

**Review by derevnjuk — CHANGES_REQUESTED** [review]

File: src/samplers/string.js [hunk 4]
```diff
@@ -105,7 +105,7 @@ const stringFormats = {
 };
 
 export function sampleString(schema) {
-    let format = schema.pattern !== undefined ? 'pattern' : schema.format || 'default';
+    let format = !isBlank(schema.pattern) ? 'pattern' : schema.format || 'default';
```
Comment (derevnjuk) [inline src/samplers/string.js]: You don't need to use `isBlank` here. Just use `schema.pattern` as condition. It's more suited for this case

```javascript
const format = schema.pattern ? 'pattern' : (schema.format || 'default');
```

**Review by derevnjuk — CHANGES_REQUESTED** [review]

**Review by derevnjuk — APPROVED** [review]

**Pull request merged**